apps/backend/src/frameworks/opencage.service.ts

```diff
@@ -0,0 +1,44 @@
+import * as https from 'https';
+
+export class OpenCageService {
+  async reverseGeocoding(latitude: string, longitude: string): Promise<any> {
+    if (!latitude || !longitude) {
+      throw new Error('Latitude and longitude must be provided.');
+    }
+
+    const url = `https://api.opencagedata.com/geocode/v1/json?q=${latitude}+${longitude}&key=${process.env.OPENCAGE_KEY}`;
+
+    return new Promise((resolve, reject) => {
+      https.get(url, (response) => {
+        let data = '';
+
+        response.on('data', (chunk) => {
+          data += chunk;
+        });
+
+        response.on('end', () => {
+          const parsedData = JSON.parse(data);
+          if (parsedData && parsedData.results) {
+            resolve(parsedData.results.map(address => {
+              if (address.components.continent == "Europe") {
+                return {
+                  region: address.components.municipality,
+                  subregion: address.components.county,
+                }
+              } else {
+                return {
+                  region: address.components.city,
+                  subregion: address.components.state,
+                }
+              }
```

Comment:
hmmm i don't know... maybe we could just use the `municipality` and `county` as simple fallbacks for everything, not just Europe? WDYT?

```js
const { 
 city,
 state, 
 municipality,
 county
} = address.components;

return {
 region: city || municipality,
 subregion: state || county
};